docs/allocations.md

```diff
@@ -0,0 +1,280 @@
+# Allocations
+
+For high-performance software in Swift, it's often important to understand where your heap allocations are coming from. The next step can then be to reduce the number of allocations your software makes.
+
+This is very similar to other performance questions: Before you can optimise performance you need to understand where you spend your resources. And resources can be CPU time, as well as memory, or heap allocations.
+In this document we will solely focus on the number of heap allocations, not their size.
+
+On macOS, you can use Instruments's "Allocations" instrument. The Allocations instrument shows you two sets of values: The live allocations (i.e. allocated and not freed) as well as the transient allocations (all allocations made).
+
+Your production workloads however will likely run on Linux and depending on your setup the number of allocations can differ significantly between macOS and Linux.
+
+## Preparation
+
+To not waste your time, be sure to do any profiling in _release mode_. Swift's optimiser will produce significantly faster code which will also allocate less in release mode. Usually this means you need to run
+
+    swift run -c release
+    
+#### Install `perf`
+
+Follow the [installation instructions](linux-perf.md) in the Linux `perf` utility guide.
+
+#### Clone the `FlameGraph` project
+
+To see some pretty graphs, clone the [`FlameGraph`](https://github.com/brendangregg/FlameGraph) repository on the machine/container where you need it. The rest of this guide will assume that it's available at `/FlameGraph`:
```

Comment:
Quite an aggressive assumed location.

Reply:
nice and short ;). Folks can always replace this with a real path, no? I do bind mount my flame graph checkout into `/FlameGraph` in all my containers :P

Reply:
I feel it didn't make much sense to come up with a certain path that is maybe more realistic because in the end, users will need to replace it with where they have it, right?